sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -311,13 +312,18 @@ public void updateAddresses(final URI serverKey) {
         headers.put(HttpConstants.HttpHeaders.X_DATE, Utils.nowAsRFC1123());
 
         if (tokenProvider.getAuthorizationTokenType() != AuthorizationTokenType.AadToken) {
-            String token = this.tokenProvider.getUserAuthorizationToken(
+            String token = null;
+            try {
+                token = this.tokenProvider.getUserAuthorizationToken(
                     collectionRid,
                     ResourceType.Document,
                     RequestVerb.GET,
                     headers,
                     AuthorizationTokenType.PrimaryMasterKey,
                     request.properties);
+            } catch (UnauthorizedException e) {
+                // User doesn't have rid based resource token. Maybe user has name based.
```

Comment:
We are swallowing the exception here. If there is no active action that we can take, we should at least log it for our own purposes.

Reply:
Make sense,  since we will retry using name based link next, so will add a debug level message to not confuse the customer

Reply:
seems to me we are throwing `UnauthorizedException` as part of the logic rather that catching a problem

Why can't we do this with a if/else condition where you check it is rid based use that else use name based.
Why do we need to throw exception can't rely on if/else?

exception throwing is expensive!

Reply:
The intention here seems like: first always try resouceId, then using named based link as fallback. so it make sure we will try both ways.

If we changed to if else, then we will only try one of them.

Currently, the main issue(query does not work with permissions) is that we are mixing using the name based link or the resouceId based link: how we construct the tokenMap and how we consuming it.  There could be two possible ways to fix it:
a. using the namedBasedLink/ResourceIdBasedLink consistently
b. try one, if not found, then fallback to another one. 

Depending which way we want to choose, this part of logic can be simplified or should keep some similar logic.